test/infrastructure/docker/internal/docker/util.go

```diff
@@ -118,3 +123,12 @@ func list(ctx context.Context, visit func(context.Context, string, *types.Node),
 
 	return nil
 }
+
+func hasCloudProviderTaint(node *corev1.Node) bool {
```

Comment:
I would suggest to move this into our internal/taints util as `HasTaint(taints []corev1.Taint, targetTaint corev1.Taint) bool` (and also call it from kubeadmconfig_controller)